Resources/config/resource_owners.xml

```diff
@@ -0,0 +1,91 @@
+        <parameter key="hwi_oauth.resource_owner.toshl.class">HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\ToshlResourceOwner</parameter>
+        <parameter key="hwi_oauth.resource_owner.trakt.class">HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\TraktResourceOwner</parameter>
+        <parameter key="hwi_oauth.resource_owner.trello.class">HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\TrelloResourceOwner</parameter>
+        <parameter key="hwi_oauth.resource_owner.twitch.class">HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\TwitchResourceOwner</parameter>
+        <parameter key="hwi_oauth.resource_owner.twitter.class">HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\TwitterResourceOwner</parameter>
+        <parameter key="hwi_oauth.resource_owner.vkontakte.class">HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\VkontakteResourceOwner</parameter>
+        <parameter key="hwi_oauth.resource_owner.windows_live.class">HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\WindowsLiveResourceOwner</parameter>
+        <parameter key="hwi_oauth.resource_owner.wordpress.class">HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\WordpressResourceOwner</parameter>
+        <parameter key="hwi_oauth.resource_owner.xing.class">HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\XingResourceOwner</parameter>
+        <parameter key="hwi_oauth.resource_owner.yahoo.class">HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\YahooResourceOwner</parameter>
+        <parameter key="hwi_oauth.resource_owner.yandex.class">HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\YandexResourceOwner</parameter>
+        <parameter key="hwi_oauth.resource_owner.odnoklassniki.class">HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\OdnoklassnikiResourceOwner</parameter>
+        <parameter key="hwi_oauth.resource_owner.37signals.class">HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\ThirtySevenSignalsResourceOwner</parameter>
+        <parameter key="hwi_oauth.resource_owner.itembase.class">HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\ItembaseResourceOwner</parameter>
+    </parameters>
+
+    <services>
+        <!-- Resource owners -->
+        <service id="hwi_oauth.abstract_resource_ownermap" class="HWI\Bundle\OAuthBundle\Security\Http\ResourceOwnerMap" abstract="true">
+            <argument key="$httpUtils" type="service" id="security.http_utils" />
+            <argument key="$possibleResourceOwners">%hwi_oauth.resource_owners%</argument>
+            <argument key="$resourceOwners" type="collection" /><!-- map -->
+            <argument key="$locator" type="tagged_locator" tag="hwi_oauth.resource_owner" index-by="resource-name" />
+        </service>
+
+        <service id="hwi_oauth.resource_ownermap_locator" class="HWI\Bundle\OAuthBundle\Security\Http\ResourceOwnerMapLocator" />
+
+        <instanceof id="HWI\Bundle\OAuthBundle\OAuth\ResourceOwnerInterface" autowire="true">
+            <tag name="hwi_oauth.resource_owner" />
+        </instanceof>
```

Comment:
I've seen usually this done in the Extension like https://github.com/symfony/symfony/blob/2cee75bef9f6c1ca4e532986dc0dcae8461de4dd/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L386-L387

Reply:
Removed totally as in fact I added it already in extension while defining the resource owner definition.